sdk/identity/identity/src/credentials/managedIdentityCredential/appServiceMsi2017.ts

```diff
@@ -28,14 +26,20 @@ function prepareRequestOptions(resource: string, clientId?: string): RequestPrep
     queryParameters.clientid = clientId;
   }
 
+  const query = new URLSearchParams(queryParameters);
+
+  // This error should not bubble up, since we verify that this environment variable is defined in the isAvailable() method defined below.
+  if (!process.env.MSI_ENDPOINT) {
+    throw new Error("Missing environment variable: MSI_ENDPOINT");
+  }
+
   return {
-    url: process.env.MSI_ENDPOINT,
+    url: `${process.env.MSI_ENDPOINT}?${query.toString()}`,
     method: "GET",
-    queryParameters,
-    headers: {
+    headers: createHttpHeaders({
       Accept: "application/json",
-      secret: process.env.MSI_SECRET
-    }
+      secret: process.env.MSI_SECRET!
```

Comment:
do you need the bang here even with the check at line 32?

Reply:
yep it doesn't like non-string values

Reply:
I think the check at line 32 will make sure it is not `undefined` so I believe TS will see it as just a string at this point.

Reply:
Nope, I get complaints from TS :(

Reply:
L22 checks MSI_ENDPOINT but this variable is MSI_SECRET. You haven't checked that this variable is defined, is that okay?

Reply:
oh let me check

Reply:
well! turns out [Python also checks for this secret to be defined](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/app_service.py#L56), so I will check for this secret to exist as well